utils/aws/route53/__init__.py

```diff
@@ -0,0 +1,318 @@
+
+    def __hash__(self):
+        return f"{self.name}".__hash__()
+
+    def __repr__(self):
+        return f"Zone<{self.name}>"
+
+
+class Record(object):
+    """
+    Record represents a DNS record
+
+    :param zone: the zone the record belongs to
+    :param name: the record name
+    :param data: the record data
+    :param awsdata: the original aws data for a dns record
+    :type zone: Zone
+    :type name: str
+    :type data: dict
+    :type awsdata: dict
+    """
+
+    SUPPORTED_TYPES = ['A', 'CNAME', 'NS', 'TXT']
+
+    def __init__(self, zone, name, data={}, awsdata={}):
+        self._zone = zone
+        self._name = _octal_replace(name).lower().rstrip('.')
+
+        try:
+            self._type = data['type']
+            self._ttl = int(data['ttl'])
+        except KeyError as e:
+            raise InvalidRecordData(f"missing key {e} in Record data")
+
+        if self._type not in self.SUPPORTED_TYPES:
+            raise InvalidRecordType(f"Type {self._type} is not supported")
+
+        self._values = data.get('values', [])
+        self._data = data
+        self._awsdata = awsdata
+
+    @property
+    def data(self):
+        """Returns the record data"""
+        return self._data
+
+    @property
+    def fqdn(self):
+        """Returns the record's FQDN"""
+        return f"{self.name}.{self.zone.name}"
+
+    @property
+    def name(self):
+        """Returns the record name"""
+        return self._name
+
+    @property
+    def awsdata(self):
+        """Returns the record's awsdata"""
+        return self._awsdata
+
+    @property
+    def ttl(self):
+        """Returns the record TTL"""
+        return self._ttl
+
+    @property
+    def type(self):
+        """Returns the record type"""
+        return self._type
+
+    @property
+    def zone(self):
+        """Returns the parent zone"""
+        return self._zone
+
+    @property
+    def values(self):
+        """Returns the values for the record"""
+        return sorted(self._values)
+
+    def add_targets(self, values):
+        """
+        Add a list of targets to the record
+
+        :param values: a list of values to add
+        :type values: list of str
+        """
+        if not isinstance(values, list):
+            raise TypeError(f"expected list, received {type(values)}")
+        self._values.extend(values)
+
+    def stringify(self):
+        """Returns a human readable representation of the record"""
+        return f"{self.name} {self.type} {self.ttl} {self.values}"
+
+    def __eq__(self, other):
+        if not isinstance(other, Record):
+            return False
+        return (self.name == other.name and
+                self.ttl == other.ttl and
+                self.type == other.type and
+                self.values == other.values)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return f"{self.name}:{self.type}:{self.ttl}:{self.values}".__hash__()
```

Comment:
Two tips here:
- use a tuple, so you never face issues with the type conversions to string.
- use `hash()` instead of `.__hash__()`. The builtins are safer than the dunders.

Reply:
removed __hash__ functions as we're not using sets anymore 8e3c392112e56cd517ab230cdd4314f532e3e5d3